### PR TITLE
Issue #896: provide a "proxyReq" event also for websocket connections.

### PR DESCRIPTION
**Pull request by lbrucher**

### CODE REVIEW

File: test/lib-http-proxy-test.js [hunk 1]
```diff
@@ -444,6 +444,49 @@ describe('lib/http-proxy.js', function() {
...
+          proxy,
+          proxyServer,
+          destiny;
+
+      proxy = httpProxy.createProxyServer({
+        target: 'ws://127.0.0.1:' + ports.source,
+        ws: true
+      });
+
+      proxy.on('proxyReqWs', function(proxyReq, req, res, options) {
```
Comment (jcrugzz) [inline test/lib-http-proxy-test.js]: slight nitpick on naming, `res` should be `socket`

File: lib/http-proxy/passes/ws-incoming.js [hunk 2]
```diff
@@ -87,6 +87,10 @@ var passes = exports;
     var proxyReq = (common.isSSL.test(options.target.protocol) ? https : http).request(
       common.setupOutgoing(options.ssl || {}, options, req)
     );
+
+    // Enable developers to modify the proxyReq before headers are sent
+    if(server) { server.emit('proxyReqWs', proxyReq, req, socket, options, head); }
```
Comment (jcrugzz) [inline lib/http-proxy/passes/ws-incoming.js]: space after the `if`

**Comment by jcrugzz** [comment]
@lbrucher Looks good, will get it in a release later today :+1:

**Comment by lbrucher** [comment]
fixed, thanks for the checks!

**Comment by andrewpmartinez** [comment]
=\

https://github.com/nodejitsu/node-http-proxy/pull/883

**Pull request merged**